app/templates/pwa/ts/custom-service-worker.ts

```diff
@@ -0,0 +1,12 @@
+/*
+ * This file (which will be your service worker)
+ * is picked up by the build system ONLY if
+ * quasar.conf > pwa > workboxPluginMode is set to "InjectManifest"
+ */
+
+import { precacheAndRoute } from 'workbox-precaching';
+
+declare const self: ServiceWorkerGlobalScope;
```

Comment:
This seems a pretty old problem: https://github.com/microsoft/TypeScript/issues/14877
Right now I resolved by using ad-hoc tsconfig, but Notification (and others) still won't be accessible from `self`.

I'd say it's not such a bad thing, so I'd prefer to avoid adding `& typeof globalThis` and make people use globals, do you have use cases where using globals isn't an option?

Eg. the example would work even as AFAIK

```
async function onPush(event: PushEvent) {
  if (!(Notification && Notification.permission === 'granted')) {
    return
  }
  // ...
}
```

Reply:
Funny enough, using `self.self.Notification` would provide intellisense, but generates an error.
Some weird stuff going on there.

Anyway, since `WorkerGlobalScope` also include `typeof globalThis`, I'm gonna add it here too as suggested